packages/-ember-data/tests/integration/store/serializer-for-test.js

```diff
@@ -42,10 +42,16 @@ module('integration/store - serializerFor', function(hooks) {
     let { owner } = this;
     /*
       serializer:-default is the "last chance" fallback and is
-      registered automatically as the json-api serializer.
-      unregistering it will cause serializerFor to return `undefined`.
+      the json-api serializer which is re-exported as app/serializers/-default.
+      here we override to ensure serializerFor will return `undefined`.
      */
-    owner.unregister('serializer:-default');
+    const lookup = owner.lookup;
+    owner.lookup = registrationName => {
```

Comment:
This test seems fundamentally impossible and invalid. Given how we are reexporting (or even before with the registration) we can **never** have `serializerFor` return undefined and therefore the assertion being tested can _basically_ never actually happen.

Reply:
It will happen in the future once `-default` is deprecated, and it could happen if we move to "install via blueprint" instead of "re-export in app dir" which is on our radar